Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/security/KeycloakLoginUrlAuthenticationEntryPoint.java

```diff
@@ -104,39 +103,46 @@ protected String buildRedirectUrlToLoginPage(
     }
 
     /**
-     * Allows subclasses to modify the login form URL that should be applicable for a
-     * given request.
-     * @param request the request
-     * @param response the response
-     * @param exception the exception
-     * @return the URL (cannot be null or empty; defaults to {@link #getLoginFormUrl()})
+     * Allows sub-classes to modify the login form URL that should be applicable for a given request.
+     *
+     * @param httpServletRequest The HTTP servlet request.
+     * @param httpServletResponse The HTTP servlet response.
+     * @param exception The exception
+     *
+     * @return The formulated URL (cannot be null or empty; defaults to {@link #getLoginFormUrl()}).
      */
     @Override
     protected String determineUrlToUseForThisRequest(
             final HttpServletRequest httpServletRequest,
             final HttpServletResponse httpServletResponse,
             final AuthenticationException exception) {
-        if (s_utiliseKeycloakSso) {
+        if (utiliseKeycloakSso) {
+            LOGGER.info("[ Utilising keycloak ]");
+
             final String loginFormPattern = getKeycloakLoginFormUrl();
+            LOGGER.info("### loginFormPattern: {}", loginFormPattern);
 
             final String keycloakRealmOfCustomer = System.getenv(ENV_KEYCLOAK_REALM_NAME_KEY);
+            LOGGER.info("keycloakRealmOfCustomer {}", keycloakRealmOfCustomer);
 
             String loginUrl = loginFormPattern.replaceAll(KEYCLOAK_REALM_PLACEHOLDER, keycloakRealmOfCustomer);
-            loginUrl = loginFormPattern.replaceFirst(FULL_RETURN_PATH_PLACEHOLDER, s_fullConfigurableReturnPath);
+            LOGGER.info("loginUrl[1] {}", loginUrl);
+            loginUrl = loginUrl.replaceAll(FULL_RETURN_PATH_PLACEHOLDER, fullConfigurableReturnPath);
+            LOGGER.info("loginUrl[2] {}", loginUrl);
 
-            LOGGER.info("\n\n>>> Resolved Keycloak loginUrl (via securityms): {}", loginUrl);
+            LOGGER.info(">>> Resolved Keycloak loginUrl (via securityms): {}", loginUrl);
 
             return loginUrl;
         } else {
-            LOGGER.info("\n\n[ Keycloak SSO turned off ]");
+            LOGGER.info("[ Keycloak SSO turned off ]");
 
             final String requestURI = httpServletRequest.getRequestURI().trim();
-            LOGGER.info("\n\nrequestURI: {}", requestURI);
+            LOGGER.info("requestURI: {}", requestURI);
 
-            if (requestURI.endsWith("/") || requestURI.endsWith("81")) {
+            if (requestURI.endsWith("/") || requestURI.endsWith("81") || requestURI.endsWith("80")) {
```

Comment:
This seems very fragile.  What if the requestURI is "http://www.classof1980", for instance?  Using java.net.URI to parse the requestURI would help.

Reply:
Rectified with Uri parsing.